src/community/vector-mosaic/vector-mosaic-rest/src/test/java/org/geoserver/vectormosaic/rest/VectorMosaicStoreControllerTest.java

```diff
@@ -0,0 +1,237 @@
+            fout.flush();
+        }
+        return vectorMosaicDatastore;
+    }
+
+    @Test
+    public void testShapefileHarvestingInMosaic() throws Exception {
+        File dir = new File("./target/empty");
+        dir.mkdir();
+        dir.deleteOnExit();
+
+        // Creating the coverageStore
+        File g1 = new File(dir, "granule1.zip");
+        g1.deleteOnExit();
+        try (FileOutputStream fout = new FileOutputStream(g1)) {
+            IOUtils.copy(getClass().getResourceAsStream("test-data/granule1_20250101.zip"), fout);
+            fout.flush();
+        }
+
+        final int length = (int) g1.length();
+        byte[] zipData = new byte[length];
+        try (FileInputStream fis = new FileInputStream(g1)) {
+            fis.read(zipData);
+        }
+
+        MockHttpServletResponse response = postAsServletResponse(
+                RestBaseController.ROOT_PATH + "/workspaces/vm/datastores/" + VMOSAIC_SHAPE_NAME
+                        + "/mosaic/file.shp?spi=default",
+                zipData,
+                "application/zip");
+        // Store is created
+        assertEquals(202, response.getStatus());
+        Properties props = new Properties();
+        try (FileInputStream in = new FileInputStream(vectorMosaicForShapefile)) {
+            props.load(in);
+        }
+
+        String firstKey = props.stringPropertyNames().iterator().next();
+        String firstValue = props.getProperty(firstKey);
+        String featureAttributes[] = firstValue.split("\\|");
+        assertEquals(
+                "POLYGON ((-80.61513697870227 37.06228260413163, -80.61513697870227 38.64689485238214, -78.02707286292329 38.64689485238214, -78.02707286292329 37.06228260413163, -80.61513697870227 37.06228260413163))",
+                featureAttributes[0]);
+        assertTrue(featureAttributes[1].endsWith("granule1_20250101.shp"));
+        assertEquals("2025-01-01T00:00:00Z", featureAttributes[2]);
+    }
+
+    @Test
+    public void testDGGSParquetHarvesting() throws Exception {
+        File dir = new File("./target/empty");
+        dir.mkdir();
+        dir.deleteOnExit();
+
+        // Creating the coverageStore
+        File g1 = new File(dir, "localsample.parquet");
+        g1.deleteOnExit();
+        try (FileOutputStream fout = new FileOutputStream(g1)) {
+            IOUtils.copy(getClass().getResourceAsStream("test-data/localsample.parquet"), fout);
+            fout.flush();
+        }
+        final int length = (int) g1.length();
+        byte[] fileData = new byte[length];
+        try (FileInputStream fis = new FileInputStream(g1)) {
+            fis.read(fileData);
+        }
+
+        MockHttpServletResponse response = postAsServletResponse(
+                RestBaseController.ROOT_PATH + "/workspaces/vm/datastores/" + VMOSAIC_PRQ_NAME
+                        + "/mosaic/file.parquet?spi=dggs&params=delegate.dbtype%3Dgeoparquet%3B",
```

Comment:
I would expect the delegate.dbtype to be part of the common parameters specified in the vector mosaic itself no?